# ts-doc-internal

Requires any TSDoc comments on internal objects to include either an `@internal` or an `@ignore` tag.

Internal objects are defined as classes, interfaces, or standalone functions that are not exported from the main entrypoint to the package and are not members of any exports from the main entrypoint (defined recursively). **Files that are specified as excluded in a `typedoc.json` file are ignored by this rule.**

By default, the main entrypoint is assumed to be `src/index.ts`. However, if your package's main entrypoint is elsewhere, you'll need to specify so in your `.eslintrc` configuration file as follows (for example, if the entrypoint is `index.ts`):

```json
{
  "settings": {
    "main": "index.ts"
  }
}
```

This rule exists to prevent inclusion of internal objects in documentation generated by tools such as [API Extractor](https://api-extractor.com/) and [TypeDoc](https://typedoc.org/).

## Examples

Assume all objects mentioned are not exported from the main entrypoint to the package.

### Good

```ts
/**
 * Other documentation
 * @internal
 */
class ExampleClass {}
```

```ts
/**
 * Other documentation
 * @ignore
 */
interface ExampleInterface {}
```

```ts
/**
 * Other documentation
 * @internal
 */
function exampleFunction(): void {}
```

### Bad

```ts
/**
 * Other documentation
 */
class ExampleClass {}
```

```ts
/**
 * Other documentation
 */
interface ExampleInterface {}
```

```ts
/**
 * Other documentation
 */
function exampleFunction(): void {}
```

## When to turn off

If you are not using TypeDoc.

## Source

Suggestion by [@chradek](https://github.com/chradek).
